fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/CreateJobInfo.java

```diff
@@ -0,0 +1,273 @@
+        IntervalUnit intervalUnit = IntervalUnit.fromString(intervalTimeUnit.toUpperCase());
+        if (intervalUnit == null) {
+            throw new AnalysisException("Invalid interval time unit: " + intervalTimeUnit);
+        }
+
+        // Check if interval unit is second and disable if not in test mode.
+        if (intervalUnit.equals(IntervalUnit.SECOND) && !Config.enable_job_schedule_second_for_test) {
+            throw new AnalysisException("Interval time unit can not be second in production mode");
+        }
+
+        timerDefinition.setIntervalUnit(intervalUnit);
+
+        // Set end time if provided.
+        endsTimeStampOptional.ifPresent(s -> timerDefinition.setEndTimeMs(TimeUtils.timeStringToLong(s)));
+
+        // Set immediate start if configured.
+        if (immediateStartOptional.isPresent() && Boolean.TRUE.equals(immediateStartOptional.get())) {
+            jobExecutionConfiguration.setImmediate(true);
+            // Avoid immediate re-scheduling by setting start time slightly in the past.
+            timerDefinition.setStartTimeMs(System.currentTimeMillis() - 100);
+            return;
+        }
+        // Set start time if provided.
+        startsTimeStampOptional.ifPresent(s -> timerDefinition.setStartTimeMs(TimeUtils.timeStringToLong(s)));
+    }
+
+    protected static void checkAuth() throws AnalysisException {
+        if (!Env.getCurrentEnv().getAccessManager().checkGlobalPriv(ConnectContext.get(), PrivPredicate.ADMIN)) {
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR, "ADMIN");
+        }
+    }
+
+    /**
+     * parse execute sql from create job stmt
+     * Some stmt not implement toSql method,so we need to parse sql from originStmt
+     */
+    private static String parseExecuteSql(String sql, String jobName, String comment) throws AnalysisException {
+        String lowerCaseSql = sql.toLowerCase();
+        String lowerCaseJobName = jobName.toLowerCase();
+        // Find the end position of the job name in the SQL statement.
+        int jobNameEndIndex = lowerCaseSql.indexOf(lowerCaseJobName) + lowerCaseJobName.length();
+        String subSqlStmt = lowerCaseSql.substring(jobNameEndIndex);
+        String originSubSqlStmt = sql.substring(jobNameEndIndex);
+        // If the comment is not empty, extract the SQL statement from the end position of the comment.
+        if (org.apache.commons.lang3.StringUtils.isNotBlank(comment)) {
+
+            String lowerCaseComment = comment.toLowerCase();
+            int splitDoIndex = subSqlStmt.indexOf(lowerCaseComment) + lowerCaseComment.length();
+            subSqlStmt = subSqlStmt.substring(splitDoIndex);
+            originSubSqlStmt = originSubSqlStmt.substring(splitDoIndex);
+        }
+        // Find the position of the "do" keyword and extract the execution SQL statement from this position.
+        int executeSqlIndex = subSqlStmt.indexOf("do");
```

Comment:
I think when using antler, we can get the sql string directly?

Reply:
I tried it, and it works, it's awesome